beater/java_attacher/set_run_as_user.go

```diff
@@ -0,0 +1,52 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build !windows
+// +build !windows
+
+package javaattacher
+
+import (
+	"fmt"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"syscall"
+)
+
+func (j *JavaAttacher) setRunAsUser(jvm *JvmDetails, cmd *exec.Cmd) error {
+	currentUser, err := user.Current()
+	if err != nil {
+		j.logger.Infof("failed to get the current user: %v", err)
```

Comment:
**issue**: the error is logged but execution continues. The method should return or avoid using `currentUser` if an error was returned.

Reply:
The error is now returned.

Reply:
@eyalkoren explained previously (don't remember if 1:1 or on the PR) that not finding the currentUser should only be logged but not be considered an error.

Reply:
It's returned here, but the caller logs it and continues to try running as the apm-server user.

Reply:
That's not optimal, it is only a failure to get the current user.
If we fail to get the current user, the default path should assume that the current user is `root` and the JVM user is not, thus trying to run as the target user. Trying to attach as `root` to a JVM ran by a non-root may fail.
Any reason why you think trying to run as target user is not preferable over running as APM Server user?

Reply:
Ah, now I understand. So if we can't get the current user, we should assume current user != target JVM user. I'll update it.

Reply:
Fixed and added a comment to clarify the intent.